app/containers/ItunesProvider/TrackDetailContainer/tests/index.test.js

```diff
@@ -6,59 +6,102 @@
+  });
+
+  it('should call dispatchTrackDetail on mount', async () => {
+    renderProvider(<TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy} />);
+    await timeout(500);
+    expect(submitSpy).toBeCalled();
+  });
+
+  it('should dispatchTrackDetail match to the props', () => {
+    const dispatch = jest.fn();
+    mapDispatchToProps(dispatch).dispatchTrackDetail({ trackDetail: 1440649635 });
+    expect(dispatch).toHaveBeenCalled();
+  });
+
+  it('should invoke dispatchTrackDetail when params.trackId is present', async () => {
+    const submitSpy = jest.fn();
+    jest.spyOn(require('react-router-dom'), 'useParams').mockReturnValue({ trackId: null });
+    renderProvider(<TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy} />);
+    await timeout(500);
+    expect(submitSpy).toHaveBeenCalledTimes(0);
+  });
+  it('should invoke dispatchTrackDetail when params.trackId is present', async () => {
+    const submitSpy = jest.fn();
+    jest.spyOn(require('react-router-dom'), 'useParams').mockReturnValue({ trackId: 10 });
+    const { TrackDetailContainerTest: TrackDetailContainer } = require('../index');
+    renderProvider(<TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy} />);
+    await timeout(500);
+    expect(submitSpy).toHaveBeenCalledTimes(1);
+  });
+  it('should call play the track on button click', () => {
+    const playBtnSpy = jest.spyOn(require('react'), 'useRef').mockReturnValue({ current: { play: jest.fn() } });
+    const { getByTestId } = renderProvider(<TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy} />);
+    const playBtn = getByTestId('play-btn');
+    fireEvent.click(playBtn);
+    expect(playBtnSpy).toHaveBeenCalled();
+  });
+  it('should pause the track on pause btn click', () => {
+    const { getByTestId, getByText } = renderProvider(
+      <TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy} />
     );
-    expect(getByTestId('genre').textContent).toBe(genre);
+    const pauseBtn = getByTestId('pause-btn');
+    fireEvent.click(pauseBtn);
+    expect(getByText('Track Paused')).toBeTruthy();
   });
-  it('should call dispatchTrackDetail', () => {
-    const { baseElement } = renderProvider(
-      <BrowserRouter>
-        <TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy} />
-      </BrowserRouter>
+  it('should forward the track 5s on forward btn click', () => {
+    const spy = jest.spyOn(require('react'), 'useRef').mockReturnValue({ current: { currentTime: 0 } });
+    const { getByTestId, getByText } = renderProvider(
+      <TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy} />
     );
-    expect(baseElement).toMatchSnapshot();
-    submitSpy();
-    expect(submitSpy).toBeCalled();
+    const forwardBtn = getByTestId('forward-btn');
+    fireEvent.click(forwardBtn);
+    expect(getByText('Track Forwarded 5s')).toBeTruthy();
+    expect(spy).toHaveBeenCalled();
```

Comment:
useRef will always be called when the component is rendered, so this test doesnn't really "test" anything right